File: cpp/largestNumber.cpp
#include <bits/stdc++.h>

using namespace std;

int compare(string x, string y) {
    string xy = x.append(y);

    string yx = y.append(x);

    return xy.compare(yx) > 0 ? 1: 0;
}

void largestNumber(vector<char> &A) {
    sort(A.begin(), A.end(), compare);
    for(int i=0;i<n;i++) {
        cout<<A[i]<<"";
    }
}

int main() {
    vector<char> A;
    largestNumber(A);
}

File: python/largestNumber.py
import functools
class Solution:
    # @param A : tuple of integers
    # @return a strings
    def compare(self, a, b):
        if a > b:
            return 1
        elif a < b:
            return -1
        else:
            return 0
    def largestNumber(self, A):
        strA = map(str, A)
        numS = ''.join(sorted(strA, key=functools.cmp_to_key(lambda a,b: self.compare(a+b, b+a)), reverse=True)).lstrip('0')
        return "0" if not numS else numS